### DISCUSSION TITLE
[MFA] Keep getting 200 status for step 2 instead of 302

### DISCUSSION THREAD

**Discussion by iqtedar123** [timdorr/tesla-api#397] [Q&A] [answered]
Trying to implement MFA login flow.
I am able to verify the passcode but when I make the request in step 2 again after verifying, it always returns a 200 response.

URL:
`[https://auth.tesla.com/oauth2/v3/authorize?client_id=ownerapi&code_challenge=hrBVzKRNAIDU%2Fls667jM%2FHS2ADOaiTZsmzmGa4RZNqE%3D&code_challenge_method=S256&redirect_uri=https%3A%2F%2Fauth.tesla.com%2Fvoid%2Fcallback&response_type=code&scope=openid%20email%20offline_access&state=qjVSYzYybkQhjjl 0x21839364`
Cookie: `tesla-auth.sid=s%3Az_Hx2Y6IYd4jYcrnk8nbbctmmo5YkE48.I%2FXrrrL7thgvncCncAiWI7GNEKgLLF8C%2F54k6lz4D9U;` (also tried without the semicolon)
Body: `transaction_id=wJxehoXg`

Any help would be appreciated. I was following the steps here: https://github.com/timdorr/tesla-api/issues/374

**Comment by iqtedar123** [comment, marked as answer]
Turns out I was not sending the cookie. Had to make sure the cookie had all values from Set-Cookie header.
For example:
`_abck=E6C10F92AEABA41407A3320CE8516AE9~-1~YAAQZr73SKWQDjZ5AQAABRsw7QbiCCepj8f4A1kApAW3LSTviU8CcuU/GQce7p7umSSrJxXcd5biBjFz4C3S1MMPmm0dn3hHlM7TIaZj0KPcqb/jleQhCF8pQM2Jy6VK8ufmHCuq9uVclYKcU7nyTJL2gj0l/khh7fjK4l7edWP8yliN7qD87F3ehRpNjWjjEgNUQaolVnnCozJ2XBPj/vvAZuVpYLD5xaATArmtJpLe9cOvZlqOrCEmn6wQpDL6PliqaFAayMfk6TRsNoQpV8fHryqreG+h2gK8KUHDWUGdI6GBr81nUL/OBzn2apDDE09syrOG69Bakk4oouHrBfNYZ1RWdZcWD2/zqjY7CEoH4ekDZc9TkQ==~-1~-1~-1; bm_sz=3AE293D9033B810CF803E3AE3DD4F11A~YAAQZr73SKSQDjZ5AQAABRsw7Qy958X90xzNIKdyINfxNfGOhRsuWs6jnwotxJLvX1V88FPprlWuKZpeP69ptwvp5Z2ySaNZj2VHKd+KfJg/IMjxOEyGrK7bBiTjuARVitpM9DeYOBhvESWAtRXfrqg8FnjtdZH3+9PLTA3889Y+HqTPrGsJfYvw81dhSJk=; tesla-auth.sid=s%3AGroZ1V_bY3aYU3wt-i60LzLw9sZbOcVI.tRLyWJHWezA0uu1TSQ23g2ENqLRfqR%2F%2B5yJxOajUZdQ`

**Reply by nikemeal** [reply]
when you post the email/pass initially (to get back the location code= part) are you just posting what you have above (body only has transaction, and cookie doesn't have the bits about expires etc).  only asking as im posting all and can't get it to do anything other than return to me the location that is identical to the one i called, not the one with `location=`

**Comment by LogixDE** [comment]
I got the same issue, but I'm pretty sure, that I send all cookies. I tried it with a c# applikation and with postman, but i always get the same issue.
